lib/kubeclient/common.rb

```diff
@@ -508,5 +553,11 @@ def http_options(uri)
 
       options.merge(@socket_options)
     end
+
+    def http_err(json_error_msg, e)
+      err_message = json_error_msg['message'] || e.message
+      error_klass = e.http_code == 404 ? ResourceNotFoundError : HttpError
```

Comment:
when using `Restclient::Request.execute` (line 254), `RestClient` doesn't throw errors on 404 / Resource Not Found. I think looking for a 404 status is the most consistent way to check that the requested resource is not found here. Other than checking for `Not Found` in the response string (which may or may not be present).